### PR TITLE
Add a new phenotype/region endpoint which returns the phenotype assoc…

### PR DESCRIPTION
**Pull request by ens-lgil**
…iations overlapping a given region

### CODE REVIEW

**Comment by magaliruffier** [comment]
any chance of some test cases for it please?

**Comment by ens-lgil** [comment]
Sorry, I will add them!

File: lib/EnsEMBL/REST/Controller/Phenotype.pm [hunk 1]
```diff
@@ -80,6 +80,45 @@ sub term: Chained('/') PathPart('phenotype/term') Args(2) ActionClass('REST') {
...
+application/json
+text/x-gff3
+
+=cut
+
+has 'max_slice_length' => ( isa => 'Num', is => 'ro', default => 1e7);
+
+sub region_GET {}
+
+sub region: Chained('/') PathPart('phenotype/region') Args(2) ActionClass('REST') {
```
Comment (nerdstrike) [inline lib/EnsEMBL/REST/Controller/Phenotype.pm]: Putting all the logic in sub region() guarantees rewrites if you want to add POST support later. It's too late for this code though.

File: lib/EnsEMBL/REST/Model/Phenotype.pm [hunk 2]
```diff
@@ -127,4 +127,76 @@ sub fetch_features{
...
+
+  my $pfs = $phenfeat_ad->fetch_all_by_Slice_with_ontology_accession($slice, $pf_type);
+
+  my %record_data;
+  my %phenotype_data;
+
+  foreach my $pf(@{$pfs}){
+
+    my $object_id = $pf->object_id; 
+    my $phe_id    = $pf->{_phenotype_id};
```
Comment (nerdstrike) [inline lib/EnsEMBL/REST/Model/Phenotype.pm]: Looks fishy. _ usually denotes internal, i.e. why are you accessing it directly?

Comment (ens-lgil) [inline lib/EnsEMBL/REST/Model/Phenotype.pm]: That's a quicker way to get the internal phenotype_id. I should write a method in the Variation API to get it more elegantly

Comment (lairdm) [inline lib/EnsEMBL/REST/Model/Phenotype.pm]: Sorry, I'm 100% with @nerdstrike on this one, setter/getters please. The extra few ops in the cpu won't noticeably change the response time. Thanks.

**Comment by nerdstrike** [comment]
Approved

**Comment by lairdm** [comment]
Nothing else jumps out...

**Pull request merged**